sammba/registration/struct.py

```diff
@@ -387,130 +418,85 @@ def anats_to_common(anat_filenames, write_dir, brain_volume,
-            warp_files.append(out_qwarp.outputs.source_warp)
-            warped_files.append(out_qwarp.outputs.warped_source)
-
-        out_tcat = tcat(in_files=warped_files,
-                        out_file='warped_2iters_heads.nii.gz')
-        out_tstat_warp_head = tstat(in_file=out_tcat.outputs.out_file,
-                                    outputtype='NIFTI_GZ')
 
-    ###########################################################################
-    # Using previous files and concatenated transforms can be exploited to
-    # avoid building up reslice errors.
-    # Warp with mini-patch
-    # In this particular case, minpatch=75 corresponds to a level of 4
-    if len(nonlinear_levels) > 2:
-        if nonlinear_minimal_patch is None:
-            nonlinear_minimal_patch = 75
-
-        for n_iter, inilev in enumerate(nonlinear_levels[2:]):
-            previous_warp_files = warp_files
-            warped_files = []
-            warp_files = []
-            for warp_file, centered_head_file in zip(previous_warp_files,
-                                                     centered_head_files):
-                suffixed_file = fname_presuffix(
-                    centered_head_file,
-                    suffix='_warped{}'.format(n_iter + 3))
-                if n_iter == len(nonlinear_levels):
-                    out_file = os.path.join(write_dir,
-                                            os.path.basename(suffixed_file))
-                else:
-                    out_file = suffixed_file
+        out_file = fname_presuffix(centered_head_file, suffix='_warped{}'.format(n_iter))
+    
+        for warp_file, centered_head_file in zip(previous_warp_files, 
+                                                 centered_head_files):
 
+            if n_iter == 0:
+                out_qwarp = qwarp(
+                    in_file=centered_head_file,
+                    base_file=out_tstat_allineated_head.outputs.out_file,
+                    noneg=True,
+                    iwarp=True,
+                    weight=nonlinear_weight,
+                    iniwarp=warp_file,
+                    inilev=0,
+                    maxlev=levels_minpatches[n_iter],
+                    out_file=out_file)
+                
+            if n_iter != 0 and n_iter < len(nonlinear_levels):
                 out_qwarp = qwarp(
                     in_file=centered_head_file,
                     base_file=out_tstat_warp_head.outputs.out_file,
-                    nmi=True,
                     noneg=True,
                     iwarp=True,
-                    weight=out_mask_tool.outputs.out_file,
-                    iniwarp=[warp_file],
-                    inilev=inilev,
-                    minpatch=nonlinear_minimal_patch,
+                    weight=nonlinear_weight,
+                    iniwarp=warp_file,
+                    inilev=levels_minpatches[n_iter - 1] + 1,
+                    maxlev=levels_minpatches[n_iter],
```

Comment:
same here, you can use `level_or_minpatch` instead of `levels_minpatches[n_iter]`